src/formatDate.js

```diff
@@ -50,7 +50,58 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateSplitted = date.split(fromFormat[3]);
+  const dateComponents = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i]) {
+      case 'DD':
+        dateComponents.day = dateSplitted[i];
```

Comment:
```suggestion
        dateComponents.DD = dateSplitted[i];
```

Better use masks as keys. You need to have all possible values: 'DD', 'MM', 'YY' and 'YYYY'